### ISSUE TITLE
Move to GitHub Actions

### ISSUE THREAD

**Issue by tdowrick** [UCL/scikit-surgeryvtk#136]
I've spent a few days wrangling with the CI, but still coming up against some issues trying to get it to work in headless mode.

Latest commits are on a branch (https://github.com/UCL/scikit-surgeryvtk/compare/add-github-actions)

Using Xvfb, QT/PySide2 stuff is working OK, as is plain VTK stuff.

The problem occurs specifically with the QVTKRenderWindowInteractor widget. I have added some minimum examples to https://github.com/UCL/scikit-surgeryvtk/blob/add-github-actions/tests/test_basic_qt_vtk.py that demonstrate where the problem comes in. CI results are [here](https://github.com/UCL/scikit-surgeryvtk/actions/runs/122566991)

See

Whenever the Start() method of QVTKRenderWindow interactor is called, a `BadWindow` error is thrown, as below.

```
tests/test_basic_qt_vtk.py::test_qapp PASSED
tests/test_basic_qt_vtk.py::test_vtk_cylinder PASSED
tests/test_basic_qt_vtk.py::test_qvtkrenderwindowinteracto_dont_start PASSED
tests/test_basic_qt_vtk.py::test_qvtkrenderwindowinteractor X Error of failed request:  BadWindow (invalid Window parameter)
  Major opcode of failed request:  2 (X_ChangeWindowAttributes)
  Resource id in failed request:  0x3
  Serial number of failed request:  7
  Current serial number in output stream:  9
```

I've exhausted any relevant stack overflow pages I could find, and tried different combinations of xvfb/Xorg/VirtualGL.

**Comment by tdowrick** [comment]
This is just for Ubuntu - solutions would also be needed for Mac & Windows.

**Comment by thompson318** [comment]
I can replicate the problem on my local Ubuntu at least. I'll see if I can make any sense of it.

**Comment by tdowrick** [comment]
I was able to get the Linux CI working, based on what Steve had done in Travis for SnappySonic, but there was still an issue with getting Windows CI working using GitHub Actions.

I eventually got to the bottom of the problem, which is that the Windows VM only supports OpenGL 1.1, but VTK needs OpenGL 2.0. See [here](http://vtk.1045678.n5.nabble.com/Failed-to-get-wglChoosePixelFormatARB-on-VM-td5747441.html) and [here](https://github.community/t/i-need-opengl-version-2-0-whereas-windows-lastest-has-opengl-version-1-1-0/17810) for relevant discussions on this.

I solved it by installing Mesa on the VM during setup which enables software OpenGL 2 rendering. [yml file here](https://github.com/UCL/scikit-surgeryvtk/blob/3344f062a01ea604a33f736d0cfb09df91f3711e/.github/workflows/ci.yml)

Mac CI is still not working. xvfb isn't available (also tried xquartz which doesn't work).
